plugins/inputs/fritzbox/fritzbox.go

```diff
@@ -0,0 +1,454 @@
+	if plugin.queryCycleCounter != 0 {
+		return nil
+	}
+	serviceClient := hosts.ServiceClient{
+		TR064Client: deviceClient,
+		Service:     service,
+	}
+	connections, err := plugin.fetchHostsConnections(&serviceClient)
+	if err != nil {
+		return err
+	}
+	for _, connection := range connections {
+		_, err = uuid.Parse(connection.RightDeviceName)
+		if err == nil {
+			continue
+		}
+		tags := make(map[string]string)
+		tags["source"] = serviceClient.TR064Client.DeviceUrl.Hostname()
+		tags["service"] = serviceClient.Service.ShortId()
+		tags["host"] = connection.RightDeviceName
+		tags["host_role"] = plugin.hostRole(connection.RightMeshRole)
+		tags["host_ap"] = connection.LeftDeviceName
+		tags["host_ap_role"] = plugin.hostRole(connection.LeftMeshRole)
+		tags["link_type"] = connection.InterfaceType
+		tags["link_name"] = connection.InterfaceName
+		fields := make(map[string]interface{})
+		fields["max_data_rate_tx"] = connection.MaxDataRateTx
+		fields["max_data_rate_rx"] = connection.MaxDataRateRx
+		fields["cur_data_rate_tx"] = connection.CurDataRateTx
+		fields["cur_data_rate_rx"] = connection.CurDataRateRx
+		acc.AddGauge("fritzbox_host", fields, tags)
+		if plugin.Log.Level().Includes(telegraf.Debug) {
+			plugin.logMetric("fritzbox_host", tags, fields)
+		}
+	}
+	return nil
+}
+
+func (plugin *Fritzbox) hostRole(role string) string {
+	if role == "unknown" {
+		return "client"
+	}
+	return role
+}
+
+func (plugin *Fritzbox) fetchHostsConnections(serviceClient *hosts.ServiceClient) ([]*mesh.Connection, error) {
+	meshListPath := &hosts.X_AVM_DE_GetMeshListPathResponse{}
+	err := serviceClient.X_AVM_DE_GetMeshListPath(meshListPath)
+	if err != nil {
+		return nil, err
+	}
+	meshListResponse, err := serviceClient.TR064Client.Get(meshListPath.NewX_AVM_DE_MeshListPath)
+	if err != nil {
+		return nil, err
+	}
+	if meshListResponse.StatusCode == http.StatusNotFound {
+		return make([]*mesh.Connection, 0), nil
+	}
+	if meshListResponse.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("failed to fetch mesh list '%s' (status: %s)", meshListPath.NewX_AVM_DE_MeshListPath, meshListResponse.Status)
+	}
+	defer meshListResponse.Body.Close()
+	meshListBytes, err := io.ReadAll(meshListResponse.Body)
+	if err != nil {
+		return nil, err
+	}
+	meshList := &mesh.List{}
+	err = json.Unmarshal(meshListBytes, meshList)
+	if err != nil {
+		return nil, err
+	}
+	return meshList.Connections(), nil
+}
+
+func (plugin *Fritzbox) logMetric(name string, tags map[string]string, fields map[string]interface{}) {
+	serializer := &influx.Serializer{}
+	err := serializer.Init()
+	if err != nil {
+		plugin.Log.Error("Failed to initialize Serializer: ", err)
+		return
+	}
+	metric := metric.New(name, tags, fields, time.Now().Round(time.Nanosecond))
+	message, err := serializer.Serialize(metric)
+	if err != nil {
+		plugin.Log.Error("Failed to serialize metric: ", err)
+		return
+	}
+	plugin.Log.Debug(string(message))
+}
```

Comment:
Note that you can just use `telegraf --test` , `processors.printer` or `outputs.file` to achieve this.

Reply:
Yes, not needed. Will remove it and any reference.

Reply:
Removed via ac19a13ae83232457c5458827462e0157fcd4a5f